src/Library/WebServer.cs

```diff
@@ -113,6 +115,8 @@ internal void ListenerCallbackAsync(IAsyncResult result)
                         ClusterIdPayloadObject payloadObject = (ClusterIdPayloadObject)serializer.ReadObject(request.InputStream);
                         Diagnostics.LogInfo(FormattableString.Invariant($"received payload with cluster id : {payloadObject.clusterId}"));
                         response.StatusCode = (int)HttpStatusCode.Accepted;
+
+                        this.publisher.UpdateClusterId(payloadObject.clusterId);
```

Comment:
All incoming requests are handled on a single thread, correct? If so, there should be no racing condition, so we're good.

Reply:
yes they are on the same thread